src/qt/editor/helpwidget.cpp

```diff
@@ -123,178 +104,363 @@ class QCHFileObserver : public FileObserver {
+        action->setShortcut(QKeySequence::Back);
+        action->setShortcutContext(Qt::WidgetWithChildrenShortcut);
+        action->setToolTip("Back");
+        centralWidget->addAction(action);
+        connect(action, &QAction::triggered, this, [this]() { helpBrowser_->backward(); });
 
-    if (!helpEngine_->setupData()) {
-        const std::string error = utilqt::fromQString(helpEngine_->error());
-        delete helpEngine_;
-        throw Exception("Failed to setup the help engine:" + error, IVW_CONTEXT);
+        connect(helpBrowser_, &QTextBrowser::backwardAvailable, action, &QAction::setEnabled);
+    }
+    {
+        auto action = toolBar->addAction(QIcon(":/svgicons/link-right.svg"), tr("&Back"));
+        action->setShortcut(QKeySequence::Back);
+        action->setShortcutContext(Qt::WidgetWithChildrenShortcut);
+        action->setToolTip("Forward");
+        centralWidget->addAction(action);
+        connect(action, &QAction::triggered, this, [this]() { helpBrowser_->forward(); });
+
+        connect(helpBrowser_, &QTextBrowser::forwardAvailable, action, &QAction::setEnabled);
+    }
+    {
+        auto action = toolBar->addAction(QIcon(":/svgicons/revert.svg"), tr("&Reload"));
+        action->setShortcut(QKeySequence::Back);
+        action->setShortcutContext(Qt::WidgetWithChildrenShortcut);
+        action->setToolTip("Reload");
+        centralWidget->addAction(action);
+        connect(action, &QAction::triggered, this, [this]() { helpBrowser_->reload(); });
     }
-
-    fileObserver_ = std::make_unique<QCHFileObserver>(helpEngine_);
-
-    onModulesDidRegister_ =
-        app->getModuleManager().onModulesDidRegister([this]() { registerQCHFiles(); });
-    onModulesWillUnregister_ =
-        app->getModuleManager().onModulesWillUnregister([this]() { fileObserver_->removeAll(); });
-
-    connect(this, &HelpWidget::visibilityChanged, this, &HelpWidget::updateDoc);
 }
 
 HelpWidget::~HelpWidget() = default;
 
-void HelpWidget::registerQCHFiles() {
-    auto app = mainwindow_->getInviwoApplication();
-    for (const auto& module : app->getModules()) {
-        const auto moduleQchFile =
-            module->getPath(ModulePath::Docs) + "/" + module->getIdentifier() + ".qch";
-        if (filesystem::fileExists(moduleQchFile)) {
-            fileObserver_->addFile(moduleQchFile);
-        }
-    }
-}
-
-void HelpWidget::showDocForClassName(std::string classIdentifier) {
-    if (!helpEngine_) return;
-    requested_ = classIdentifier;
-    updateDoc();
+void HelpWidget::showDocForClassName(std::string_view classIdentifier) {
+    helpBrowser_->setCurrent(classIdentifier);
 }
 
 void HelpWidget::resizeEvent(QResizeEvent* event) {
     helpBrowser_->reload();
     InviwoDockWidget::resizeEvent(event);
 }
 
-void HelpWidget::updateDoc() {
-    if (current_ == requested_) return;
+namespace {
 
-    if (visibleRegion().isEmpty()) return;
-    current_ = requested_;
+constexpr std::string_view css = R"(
```

Comment:
using `[QTextDocument::defaultStylesheet()](http://doc.qt.io/qt-5/qtextdocument.html#-prop)` it should be possible to load the global qss stylesheet instead.

Reply:
Don't even think I want that. That is for widgets, this is not.